### DISCUSSION TITLE
References between issues and discussions

### DISCUSSION THREAD

**Discussion by okuramasafumi** [community/community#3548] [Discussions]
Currently we can refer to issues from discussions but cannot refer to discussions.
It's often the case when we start talking about the specific problem in Issues and soon we find ourselves discussing how we solve it. In this kind of situation, we'd be happy if we start discussions from issues and refer back the result of the discussion to the issue.
This is partly because discussions support threading and answering while issues don't.

It'd be ideal if we could create issues from discussions and create discussions from issues.

**Comment by PranamBhat** [comment]
Hi @okuramasafumi 

Since "Discussions" and "Issues" are totally different. Both has different meaning when it comes to problems. So better to use them accordingly.

**Reply by okuramasafumi** [reply]
That's the point. We start creating an issue and after some time notice we actually want to use discussions!

**Reply by PranamBhat** [reply]
This will help you [https://github.com/github/feedback/discussions/2952](https://github.com/github/feedback/discussions/2952)

**Reply by okuramasafumi** [reply]
Thanks, but I'd rather want to connect issues and discussions.

**Comment by PranamBhat** [comment]
Got your point. at the moment only these feature's are available : 

Reference : 


[https://docs.github.com/en/discussions/quickstart](https://docs.github.com/en/discussions/quickstart)

[https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-conversations-on-github](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-conversations-on-github)

[https://news.ycombinator.com/item?id=22388639](https://news.ycombinator.com/item?id=22388639)

**Comment by danielorbach** [comment]
Sounds like a duplicate of:
- #2971

**Comment by pacifiquem** [comment]
Any updates here?

**Comment by Hannan-10** [comment]
I think bidirectional references would improve traceability without changing the purpose of either feature.

A workflow that would work well is:

* During implementation, open an Issue to track actionable work.
* If the conversation becomes broad (design decisions, alternatives, RFCs), convert or link it to a Discussion.
* Once a decision is made, allow the Discussion to reference the original Issue so anyone reading later can see both the implementation and the reasoning behind it.

This would be especially useful for long-running features where multiple design discussions lead to a single implementation. Today it's easy to navigate from an Issue to related Discussions, but not always the other way around, which makes historical context harder to discover.

Even a simple "Related Issues" section on Discussions, similar to how pull requests close issues, would make navigating project history much easier.
